personalities/sctfe/storage.go

```diff
@@ -78,3 +87,65 @@ func (cts *CTStorage) AddIssuerChain(ctx context.Context, chain []*x509.Certific
 	}
 	return nil
 }
+
+// cachedIssuerStorage wraps an IssuerStorage, and keeps a copy the sha256 of certs it contains.
+//
+// This is intended to make querying faster. It does not keep a copy of the certs, only sha256.
+// Only up to N keys will be stored locally.
+// TODO(phboneff): add monitoring for the number of keys
+type cachedIssuerStorage struct {
+	m map[string]bool
```

Comment:
`map` is not safe for concurrent use.

Reply:
Would it make sense to use something like https://github.com/hashicorp/golang-lru ?
Might be overkill, OTOH it should degrade more gracefully if we do bump into the limits.

Reply:
Hm - I did not page it would actually *fail* on concurrent writes, thanks! I had made the assumption that we'd only write the same value so concurrent writes wouldn't matter. I could use sync.Map, or a regular map with just a mutex on writes.

LRU: I did consider it and concluded it was not necessary / misleading.  Given the math above, it's extremely unlikely that we'll ever run into the limit, so I thought it was better to be explicit about it. Also, I don't necessarily think that a lru would improve things: what matters here is the read frequency, not the write order. I think that  https://github.com/hashicorp/golang-lru only updates the frequency when you Get an item, not when you check for its existence. Last, it's reasonable to assume that operators are meant to restart their binary every week. The goal of caching is to speed things up and not make pricey calls to GCS. If Add requests fill 1 << 20 entries of caching within a week.... I think we should just rethink it from the ground up, and probably store intermediates in the same way that we do deduplication.